.github/workflows/rust.yml

```diff
@@ -2,23 +2,29 @@ name: Rust
 
 on:
   push:
-    branches: [ "master" ]
+    branches: ["master"]
   pull_request:
-    branches: [ "master" ]
+    branches: ["master"]
 
 env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  build:
-
-    runs-on: ubuntu-latest
+  ci:
+    name: ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
```

Comment:
I didn't see it last time, but can you remove the matrix for checking on all targets, since there is no platform dependent code and macos runs are expensive, I don't think this is necessary.